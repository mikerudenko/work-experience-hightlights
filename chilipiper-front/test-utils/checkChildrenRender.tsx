import React from 'react'
import { shallow } from 'enzyme'

type CheckChildrenRender = <P extends {}>(TestSubject: React.ComponentType<P>, props: P) => void

export const checkChildrenRender: CheckChildrenRender = (TestSubject, props) => {
    const Dummy = () => null

    const subject = shallow(
        <TestSubject {...props}>
            <Dummy />
        </TestSubject>
    )
    expect(subject.find(Dummy)).toHaveLength(1)
}
