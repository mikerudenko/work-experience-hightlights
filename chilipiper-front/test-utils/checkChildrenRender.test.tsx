import React from 'react'
import { checkChildrenRender } from './checkChildrenRender'

describe('test-utils', () => {
    describe('checkChildrenRender', () => {
        it('should pass if component renders its children', () => {
            expect(() => {
                const Foo: React.FunctionComponent = ({ children }) => <>{children}</>
                const props = {}

                checkChildrenRender(Foo, props)
            }).not.toThrow()
        })

        it('should fail if component does not render its children', () => {
            expect(() => {
                const Foo: React.FunctionComponent = () => <></>
                const props = {}

                checkChildrenRender(Foo, props)
            }).toThrowErrorMatchingSnapshot()
        })
    })
})
