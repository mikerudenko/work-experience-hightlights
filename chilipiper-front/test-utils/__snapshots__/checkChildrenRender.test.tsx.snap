// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-utils checkChildrenRender should fail if component does not render its children 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toHaveLength([22m[32mlength[39m[2m)[22m

Expected value to have length:
  [32m1[39m
Received:
  [31m{Symbol(enzyme.__root__): {Symbol(enzyme.__root__): [Circular], Symbol(enzyme.__unrendered__): <Foo><Dummy /></Foo>, Symbol(enzyme.__renderer__): {\\"batchedUpdates\\": [Function batchedUpdates], \\"checkPropTypes\\": [Function checkPropTypes], \\"getNode\\": [Function getNode], \\"render\\": [Function render], \\"simulateError\\": [Function simulateError], \\"simulateEvent\\": [Function simulateEvent], \\"unmount\\": [Function unmount]}, Symbol(enzyme.__node__): {\\"instance\\": null, \\"key\\": undefined, \\"nodeType\\": \\"function\\", \\"props\\": {}, \\"ref\\": null, \\"rendered\\": null, \\"type\\": Symbol(react.fragment)}, Symbol(enzyme.__nodes__): [{\\"instance\\": null, \\"key\\": undefined, \\"nodeType\\": \\"function\\", \\"props\\": {}, \\"ref\\": null, \\"rendered\\": null, \\"type\\": Symbol(react.fragment)}], Symbol(enzyme.__options__): {\\"adapter\\": {\\"options\\": {\\"enableComponentDidUpdateOnSetState\\": true, \\"legacyContextMode\\": \\"parent\\", \\"lifecycles\\": {\\"componentDidUpdate\\": {\\"onSetState\\": true}, \\"getChildContext\\": {\\"calledByRenderer\\": false}, \\"getDerivedStateFromError\\": true, \\"getDerivedStateFromProps\\": {\\"hasShouldComponentUpdateBug\\": false}, \\"getSnapshotBeforeUpdate\\": true, \\"setState\\": {\\"skipsComponentDidUpdateOnNullish\\": true}}}}, \\"disableLifecycleMethods\\": true, Symbol(enzyme.__providerValues__): undefined}, Symbol(enzyme.__providerValues__): Map {}}, Symbol(enzyme.__unrendered__): null, Symbol(enzyme.__renderer__): {\\"batchedUpdates\\": [Function batchedUpdates], \\"checkPropTypes\\": [Function checkPropTypes], \\"getNode\\": [Function getNode], \\"render\\": [Function render], \\"simulateError\\": [Function simulateError], \\"simulateEvent\\": [Function simulateEvent], \\"unmount\\": [Function unmount]}, Symbol(enzyme.__node__): undefined, Symbol(enzyme.__nodes__): [], Symbol(enzyme.__options__): {\\"adapter\\": {\\"options\\": {\\"enableComponentDidUpdateOnSetState\\": true, \\"legacyContextMode\\": \\"parent\\", \\"lifecycles\\": {\\"componentDidUpdate\\": {\\"onSetState\\": true}, \\"getChildContext\\": {\\"calledByRenderer\\": false}, \\"getDerivedStateFromError\\": true, \\"getDerivedStateFromProps\\": {\\"hasShouldComponentUpdateBug\\": false}, \\"getSnapshotBeforeUpdate\\": true, \\"setState\\": {\\"skipsComponentDidUpdateOnNullish\\": true}}}}, \\"disableLifecycleMethods\\": true, Symbol(enzyme.__providerValues__): undefined}, Symbol(enzyme.__rootNodes__): [{\\"instance\\": null, \\"key\\": undefined, \\"nodeType\\": \\"function\\", \\"props\\": {}, \\"ref\\": null, \\"rendered\\": null, \\"type\\": Symbol(react.fragment)}], Symbol(enzyme.__providerValues__): null}[39m
received.length:
  [31m0[39m"
`;
